src/model/gateway.rs

```diff
@@ -375,7 +375,7 @@ pub struct ShardInfo {
 
 impl ShardInfo {
     #[must_use]
-    pub(crate) fn new(id: ShardId, total: u32) -> Self {
+    pub fn new(id: ShardId, total: u32) -> Self {
```

Comment:
If `ShardInfo` no longer becomes part of the public API, this should be reverted too.

Reply:
Eh, I don't see why. The fields are already public.